apps/browser/src/components/Chat.tsx

```diff
@@ -164,8 +169,39 @@ const Chat: React.FC<ChatProps> = ({ evo, onMessage, messages, goalEnded, onSide
   };
 
   const handleStart = async () => {
+    if (!loadedOpenAiApiKey && !session?.user) {
+      setSignInModalOpen(true);
+      return;
+    }
+
+    if (!loadedOpenAiApiKey) {
+      const getPromptRequest = await fetch(
+        `/api/supabase/prompts?email=${session?.user?.email}`
+      );
+      const { prompts } = await getPromptRequest.json();
+      if (prompts?.length && prompts.length >= PROMPTS_CAP) {
+        const capReached = setCapReached();
+        if (capReached) return;
+      } else {
+        const addPromptRequest = await fetch(`/api/supabase/prompts`, {
+          method: "POST",
+          body: JSON.stringify({
+            email: session?.user?.email,
+            message,
+          }),
+          headers: {
+            "Content-Type": "application/json"
+          }
+        });
+
+        if (!addPromptRequest.ok) {
+          console.log("Error trying to add prompt request");
+          return
+        }
+      }
+    }
     handleSend();
-  }
+  };
```

Comment:
I don't think this verification should be sone at the component level; it would require us to cover all possible UI interactions that submit a prompt and makes us need to verify that everytime we change the UI.

This should be done at the LLM classes level; the "proxy" ones that you have. That way, all UI interactions that need this verification are funneled through a single place; where we would have the verification